tests/integration/testers/service-mesh-tester/charmcraft.yaml

```
@@ -13,6 +13,14 @@ bases:
         channel: "20.04"
 
 config:
+  options:
+    auto-join-mesh:
+      type: boolean
+      default: true
+      description: |
+        Automatically join the mesh by applying relevant labels to the charm's K8s resources on 
+        relating via the service_mesh interface
```

Since this is just a tester for our own use, we can be more direct about what this is for in this charm.  What about:

```suggestion
      description: |
        Used for `ServiceMeshConsumer`'s `auto_join` argument, which controls whether this charm will add itself to the mesh if there is a service_mesh relation established to a beacon.
```

Feel free to use your own words though :)